dds/DCPS/NetworkAddress.h

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_NETWORK_ADDRESS_H
+#define OPENDDS_DCPS_NETWORK_ADDRESS_H
+
+#include "dcps_export.h"
+
+#ifndef ACE_LACKS_PRAGMA_ONCE
+#  pragma once
+#endif /* ACE_LACKS_PRAGMA_ONCE */
+
+#include "Definitions.h"
+#include "PoolAllocator.h"
+
+#include "ace/INET_Addr.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+class OpenDDS_Dcps_Export NetworkAddress {
+public:
+  NetworkAddress();
+  NetworkAddress(const NetworkAddress& addr);
+
+  explicit NetworkAddress(const char*);
+  NetworkAddress(uint16_t port, const char*);
+
+#if defined (ACE_HAS_WCHAR)
+  explicit NetworkAddress(const wchar_t*);
+  NetworkAddress(uint16_t port, const wchar_t*);
+#endif
+
+  explicit NetworkAddress(const ACE_INET_Addr& addr);
+
+  NetworkAddress& operator=(const NetworkAddress& rhs);
+  NetworkAddress& operator=(const ACE_INET_Addr& rhs);
+
+  bool operator==(const NetworkAddress& rhs) const;
+  bool operator!=(const NetworkAddress& rhs) const;
+
+  bool operator<(const NetworkAddress& rhs) const;
+
+  bool addr_bytes_equal(const NetworkAddress& rhs) const;
+
+  ACE_INET_Addr to_addr() const;
+  void to_addr(ACE_INET_Addr&) const;
+
+  int16_t get_type() const;
+  void set_type(int16_t type);
```

Comment:
I think ACE_INET_Addr just uses `u_short` but I don't even know where that comes from. Looking into it... it looks like int16_t and u_short are defined in the same ACE header... `os_include/os_stdint.h`. I don't have a strong opinion here, other than desiring to avoid the identifier CORBA if possible.

Reply:
A few options to consider:
- this is a public API of a wrapper type, just use plain `int` -- it's not a protocol "on the wire" type -- and convert internally
- use whatever ACE defines in its os_include (but this is a bit messy since it wants to make POSIX and nonstandard unixy types interface types on all platforms)
- use ACE's CDR_Base types

Reply:
CDR_Base types it is...